### PR TITLE
Update to JuMP v0.23 and switch from Cbc to HiGHS

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

File: test/functions.jl [hunk 1]
```diff
@@ -13,45 +13,44 @@
 
     @objective(m, Max, dot(v,x))
 
-    @NLconstraint(m, sum(w[i]*x[i]^2 for i=1:5) <= 45)   
+    @NLconstraint(m, sum(w[i]*x[i]^2 for i=1:5) <= 45)
 
     MOIU.attach_optimizer(m)
-    bm = JuMP.backend(m)
-    options = bm.optimizer.model.optimizer.options
+    options = JuMP.unsafe_backend(bm).options
```
Comment (odow) [inline test/functions.jl:19]:
```suggestion
    options = JuMP.unsafe_backend(m).options
```

**Comment by odow** [comment]
It's nice to see that HiGHS passes all the tests.

File: test/functions.jl [hunk 2]
```diff
@@ -13,45 +13,44 @@
 
     @objective(m, Max, dot(v,x))
 
-    @NLconstraint(m, sum(w[i]*x[i]^2 for i=1:5) <= 45)   
+    @NLconstraint(m, sum(w[i]*x[i]^2 for i=1:5) <= 45)
 
     MOIU.attach_optimizer(m)
-    bm = JuMP.backend(m)
-    options = bm.optimizer.model.optimizer.options
+    options = JuMP.unsafe_backend(m).options
```
Comment (ccoffrin) [inline test/functions.jl:19]: It looks to me this was the only breaking change that impacted Juniper in the move from MOI v0.10 to 1.0? Is that correct?

**Review by ccoffrin — APPROVED** [review]

File: test/functions.jl (hunk 2 above)
Comment (odow) [reply test/functions.jl:19]: It's a change in how JuMP manages its backend. 

I don't think we ever encouraged people to call `.optimizer.model.optimizer`, so this is arguably a bug-fix. We should have used `direct_model` if we cared about the backend. The new `unsafe_backend` provides a nice way to get the inner model for cases like this.

**Pull request merged**